### DISCUSSION TITLE
Rotor 37 compressor tutorial {error}

### DISCUSSION THREAD

**Discussion by Mohsin-Bashir** [mdolab/dafoam#542] [Tutorials]
Hi I tried running Rotor 37 compressor tutorial but I am getting the following error:

```
Driver debug print for iter coord: rank0:pyOptSparse_IPOPT|46
-------------------------------------------------------------
Design Vars
{'dvs.shapey': array([-0.0012010487224063, -0.0011934035121023,  0.0001699394266906,
        0.0008782498538173,  0.0003870026181963,  0.0001303548338736,
        0.0012197865696925,  0.0017237651090031,  0.0001514866614879,
       -0.0012464622218164, -0.0013257691752231, -0.0001739719393784,
        0.0004262111891617,  0.0008660755950834, -0.0006579723394484,
       -0.0007531755630149,  0.0009830029772679, -0.0003626503651015,
        0.001018220241175 ,  0.0003338325504884, -0.0008159949227372,
        0.001158688796741 , -0.0007870852507845,  0.0000259046223882,
       -0.0010689189046014, -0.0011763247093104, -0.0011834052627649,
       -0.0011360929006465,  0.0013039181327441,  0.0011155292573585]),
 'dvs.shapez': array([ 0.0015460422136698, -0.0001306064637271, -0.0003301537417101,
       -0.0011793901686351,  0.0000294421494105, -0.0012213421488421,
       -0.0010373961322752, -0.0019825208033037,  0.0007204120028872,
        0.0011768850977943,  0.0011606459849554, -0.0009110319062096,
       -0.0010702264213511, -0.0010266856016886,  0.0005675406631697,
        0.0006061242048277,  0.0010542606426683,  0.0012816951207543,
       -0.0010183577830961, -0.0004304589029808, -0.0010755723204621,
       -0.0013148808841092, -0.0000941353541401, -0.0009695717602545,
        0.0011773981018751,  0.0014966483655424,  0.0013420889056823,
        0.0012424647534775, -0.0015402965231411, -0.0014180443610284])}

time="2023-12-20T19:56:52+08:00" level=error msg="error waiting for container: "
```

what could be the issue?
I ran the first NACA0012 tutorial and I did not face any issue.

**Comment by Mohsin-Bashir** [comment]
Ok I ran it again and got another error. But on third try, I think I ran it successfully, though i did not get any message that the simulation ran successfully (something like that). Here are the last few lines of simulation run: 

```
Nonlinear constraints
{'cruise.aero_post.functionals.MFR': array([0.6999547674215936]),
 'cruise.aero_post.functionals.TPR': array([1.5996781839952374])}

Objectives
{'cruise.aero_post.functionals.CMZ': array([17.037434903734535])}

Computing partials for  ['CMZ']
Calculating dFdW using reverse-mode AD
    Adding all faces of patches 1(blade) ...
    Found matching patch blade with 465 faces.
Calculating dFdXv using reverse-mode AD
    Adding all faces of patches 1(blade) ...
    Found matching patch blade with 465 faces.
Moving time 2120 to 0.00000018
Driver total derivatives for iteration: 18
---------------------------------------------
Computing dRdWTPC 15758 s
Initializing dRdWCon. 15758 s
dRdWCon Created. 15767 s
Reading Coloring dRdWColoring_6
Validating Coloring...
No Conflicting Colors Found!
Partial derivative matrix created. 15768 s
dRdWTPC: 0 of 1369, ExecutionTime: 15768 s

D:\tutorials-main\Rotor37_Compressor>
```

I have checked the "paraview.foam" file and results seem to be different than the ones given here "https://dafoam.github.io/mydoc_tutorials_aero_rotor37.html".

I am using windows (docker desktop) to run this simulation

**Comment by friedenhe** [comment]
I'm not sure what caused the issue you had. Maybe you need more memory... But it seems that the optimization ran for 18 iterations. Please check: https://dafoam.github.io/mydoc_get_started_post_processing.html on how to post-process the results. How different are your results compared with the figures on the R37 tutorial website?

**Comment by Mohsin-Bashir** [comment]
Results for pressure gradient seem to be different, Following picture shows my results
![myResults](https://github.com/mdolab/dafoam/assets/118594705/12563a7d-88d0-4011-9cec-32639c45c496)
Following picture shows R37 tutorial website results
![r37_tutorial_results](https://github.com/mdolab/dafoam/assets/118594705/4c237fca-c6e4-4383-96ea-0f14a910c8d8)
 This time optimization ran for 53 iterations  and terminated without giving any messages about completion.

My PC's specs are:
Processor: 13th Gen Intel(R) Core(TM) i5-13400   2.50 GHz
Installed RAM: 24.0 GB

I ran this optimization in parallel using 4 cores. May be its because of less memory. I shall try to run this on workstation later.

**Comment by friedenhe** [comment]
OK, we recently changed the boundary condition for the Rotor 37 case. I used fixedValue for the T and P, and we changed them to totalPressure and totalTemperature. So, the working conditions may have changed due to this case. I suspect this caused the difference you saw. As long as your objective function is decreasing, you should be good.
